### PR TITLE
tools deprecation and new format support

### PR DESCRIPTION
**Pull request by tarun-aiplanet**

### CODE REVIEW

File: src/openagi/actions/tools/arxiv_search.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
+from openagi.actions.base import BaseAction
+from pydantic import Field
+from openagi.exception import OpenAGIException
+from typing import ClassVar, Dict, Any
+
+try:
+    import arxiv
```
Comment (gourab-aiplanet) [inline src/openagi/actions/tools/arxiv_search.py:7]: If we import this one will it work or explicitly we will have to run the install command? @tarun-aiplanet

File: src/openagi/actions/tools/exasearch.py [hunk 2]
```diff
@@ -1,30 +1,75 @@
...
 
 try:
-   from exa_py import Exa
+    from exa_py import Exa
 except ImportError:
-  raise OpenAGIException("Install Exa Py with cmd `pip install exa_py`")
+    raise OpenAGIException("Install Exa Py with cmd `pip install exa_py`")
 
-class ExaSearch(BaseAction):
+class ConfigurableAction(BaseAction):
```
Comment (gourab-aiplanet) [inline src/openagi/actions/tools/exasearch.py:13]: @tarun-aiplanet why are we redefining the ConfigurableAction class?

File: src/openagi/actions/tools/arxiv_search.py (hunk 1 above)
Comment (tarun-aiplanet) [reply src/openagi/actions/tools/arxiv_search.py:7]: we will have to run install command

File: src/openagi/actions/tools/google_search_tool.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import requests
+from bs4 import BeautifulSoup
+from typing import ClassVar, Dict, Any
+
+try:
+   from googlesearch import search
+except ImportError:
+  raise OpenAGIException("Install googlesearch-python with cmd `pip install googlesearch-python`")
+
+class ConfigurableAction(BaseAction):
```
Comment (gourab-aiplanet) [inline src/openagi/actions/tools/google_search_tool.py:14]: Define it separately and use import in all the places. @tarun-aiplanet

File: src/openagi/actions/tools/exasearch.py (hunk 2 above)
Comment (tarun-aiplanet) [reply src/openagi/actions/tools/exasearch.py:13]: we need this for configuring the api keys and other user defined params right?

**Review by gourab-aiplanet — APPROVED** [review]

**Pull request merged**